src/async_impl/h3_client/mod.rs

```diff
@@ -73,6 +95,42 @@ impl H3Client {
             .map_err(|e| Error::new(Kind::Request, Some(e)))
     }
 
+    #[cfg(feature = "cookies")]
```

Comment:
Hm, this makes me wonder if we could turn this into some internal cookies middleware that we apply to both. Not to worry for this PR, but perhaps a follow-up, if it makes sense.

Reply:
It does make sense to turn `cookies` logic into an middleware. You are right, maybe later I or anyone can put this into follow-up PR.

Reply:
Alright, I put this in an issue :) https://github.com/seanmonstar/reqwest/issues/2693